src/utils/page-functions.js

```diff
@@ -0,0 +1,202 @@
+import { createHash } from 'crypto'
+import path from 'path'
+import webpack from 'webpack'
+import { camelCase } from '../utils'
+import {
+  readFile,
+  exists,
+  stats,
+  requireResolve
+} from './fs'
+
+let packageName
+let nodeModulesPath
+
+/* export async function listPageFunctionSources(fnPath) {
+  if (!fnPath) {
+    fnPath = path.join(__dirname, 'page-functions')
+  }
+
+  const files = await readDir(fnPath, { encoding: 'utf8' })
+  return files.filter(file => file.endsWith('.js'))
+} */
+
+export function createPageFunctions(page, sourceFiles) {
+  const pageFunctions = {}
+
+  if (!sourceFiles) {
+    return pageFunctions
+  }
+
+  const babelPresets = page.getBabelPresetOptions()
+
+  for (let file of sourceFiles) {
+    let fnName
+    if (Array.isArray(file)) {
+      [file, fnName] = file
+    } else {
+      fnName = camelCase(path.basename(file, '.js'))
+    }
+
+    pageFunctions[fnName] = async (...args) => {
```

Comment:
Should we check for fnName conflicts? sourceFiles may contain files with the same name but from different directories. Maybe we can simply suffic with a number or warn or ...

Reply:
These fnName's are the names you need to use to call them on the page object in your tests. So renaming them could be problematic as well. Will add a warning for now.